interface/src/app/download/download.spec.js

```diff
@@ -2,21 +2,113 @@
  * Unit tests.
  */
 
-describe('DownloadCtrl', function() {
-  describe('isCurrentUrl', function() {
-    var DownloadCtrl, $location, $scope;
-
-    beforeEach(module('adage'));
-
-    beforeEach(inject(function($controller, _$location_, $rootScope) {
-      $location = _$location_;
-      $scope = $rootScope.$new();
-      DownloadCtrl = $controller('DownloadCtrl',
-                                 {$location: $location, $scope: $scope});
-    }));
-
-    it('validates DownloadCtrl controller', inject(function() {
-      expect(DownloadCtrl).toBeTruthy();
-    }));
+// Test the generic controller properties (without server calls):
+describe('DownloadCtrl w/o server calls', function() {
+  beforeEach(module('adage.download'));
+
+  var ctrl, mockService;
+  // Create a mocked "AnnotationTypes" service, which overwrites the service
+  // in download.js (because it is loaded after "adage.download" module).
+  beforeEach(module(function($provide) {
+    mockService = {
+      get: function() { }
+    };
+    $provide.value('AnnotationTypes', mockService);
+  }));
+
+  beforeEach(inject(function($controller) {
+    ctrl = $controller('DownloadCtrl');
+  }));
+
+  it('should be defined', function() {
+    expect(ctrl).toBeDefined();
```

Comment:
After putting more thoughts on this, I believe that here `expect(ctrl).tpBeDefined()` and  `expect(ctrl).tpBeTruthy()` are almost equivalent. When some error prevents the controller instance from being instantiated, it wouldn't be defined, and it would not be truthy (but `truthy` is more general than `true`) either, so for an object this is equivalent. For a string like this:
```var foobar = '';```
then `foobar` is defined but it is not truthy.